src/Libraries/Microsoft.Extensions.AI/ChatCompletion/CachingChatClient.cs

```diff
@@ -178,4 +195,13 @@ public override async IAsyncEnumerable<ChatResponseUpdate> GetStreamingResponseA
     /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
     /// <exception cref="ArgumentNullException"><paramref name="value"/> is <see langword="null"/>.</exception>
     protected abstract Task WriteCacheStreamingAsync(string key, IReadOnlyList<ChatResponseUpdate> value, CancellationToken cancellationToken);
+
+    /// <summary>Determine whether to use caching with the request.</summary>
+    private static bool UseCaching(ChatOptions? options)
+    {
+        // We want to skip caching if options.ConversationId is set. If it's set, that implies there's
+        // some state that will impact the response and that's not represented in the messages. Since
+        // that state could change even with the same ID, we have to assume caching isn't valid.
```

Comment:
Interesting - this would be something to keep in mind if we see any reports around response caching not working in the context of eval (cc @peterwald).

Evaluators themselves are unlikely to specify ConversationId, so this is unlikely to impact caching for the evaluation / scoring responses. But this could impact caching of the model responses being evaluated (since it is possible that the ChatOptions.ConversationId can be specified somewhere in the product / agent code, and this would turn off the caching even if the test code were passing down a caching IChatClient when fetching the agent response that is to be evaluated).

Just noting as something to double-check / keep in mind in case we hear any reports around eval caching not working as intended :)